README.md

```diff
@@ -1,10 +1,78 @@
 # proofcarryingdata/snark-artifacts
 
 Repo for holding large binaries needed for ZK proofs, particularly for families
-of GPCs (General Purpose Circuits).  These are kept out of the `zupass` monorepo
-to avoid bloat.
+of GPCs (General Purpose Circuits).  These binaries are used by code in the
+[Zupass](https://github.com/proofcarryingdata/zupass) repo, but stored in a
+separate repo to avoid bloat.  For more details on what is available,
+see the individual packages in the `packages/` directory.
 
-Everything here is still experimental while the specific method of artifact
-generation and distribution is being worked out.
+## Downloading artifacts
 
-TODO(POD-P3): Document how to find and use artifacts.
+For testing, you can clone this repo directly to obtain artifacts.  Artifacts
+can also downloaded directly in a browser from this repo using a URL like the
+example below, where "rev" can be a git branch, tag, or commit hash.
+
+  https://raw.githubusercontent.com/proofcarryingdata/snark-artifacts/rev/packages/proto-pod-gpc/proto-pod-gpc_1o-5e-6md-0x0l-0x0t-vkey.json
+
+Officially released artifacts are available as NPM packages accessible as
+devDependencies.  They can also be downloaded from there using an NPM-based
+CDN such as unpkg.com:
+
+  https://unpkg.com/@pcd/proto-pod-gpc-artifacts@0.0.2/proto-pod-gpc_1o-5e-6md-0x0l-0x0t-vkey.json
+
+The `@pcd/gpc` package in the Zupass needs a root URL to find artifacts by
+either of these download mechanism.  To create an appropriate URL you can
+use the `gpcArtifactDownloadURL` function available
+[here](https://github.com/proofcarryingdata/zupass/blob/main/packages/lib/gpc/src/gpc.ts).
+
+## Testing experimental artifacts
+
+The direct GitHub download method above can be used for testing with
+experimental artifacts during circuit development.  These commands assume
+you've cloned the `zupass` and `snark-artifacts` directories in a single
+parent folder and have write access to each.
+
+  cd zupass/packages/lib/gpcircuits
+  yarn gen-test-artifacts
+  yarn copy-test-to-snark-artifacts
+  cd ../../../../snark-artifacts
+  git checkout -b your-experimental-branch-name
+  git commit . -m "test artifacts"
+  git push
+
+You can then use `your-experimental-branch-name` as the rev in download
+URLs above.
```

Comment:
for ppl who don't have write access to this repo, the url will need to refer to a different repo (as PRs for external contributors will be coming from forks, not from this repo. not a huge issue as this readme is probably only intended for internal use anyways.